### PR TITLE
[IMP] base: add flexibility for attibutes inheritance

### PR DESCRIPTION
**Pull request by Gorash**
Can use 'append', 'prepend' or 'eval' on attribute tag.
E.g.: <attribute name="class" eval="value.replace('o_red', 'o_green')"/>

### CODE REVIEW

**Comment by wtaferner** [comment]
:+1:

File: openerp/addons/base/ir/ir_ui_view.py [hunk 1]
```diff
@@ -529,11 +529,19 @@ def apply_inheritance_specs(self, cr, uid, source, specs_tree, inherit_id, conte
...
-                            del node.attrib[attribute[0]]
+                        attribute = child.get('name')
+                        value = child.text or None
+                        if child.get('add') or child.get('remove'):
+                            value = node.get(attribute, "")
+                            assert not child.text
+                        if child.get('add'):
+                            value = "%s %s" % (value, child.get('add').strip())
+                        if child.get('remove'):
+                            value = " ".join([val for val in value.split(" ") if (" %s " % val) not in (" %s " % child.get('remove'))])
```
Comment (xmo-odoo) [inline openerp/addons/base/ir/ir_ui_view.py]: Why not just use set operations there? e.g. 
```python
to_add = set(child.get('add', '').split())
to_remove = set(child.get('remove', '').split()
value = ' '.join(set(value.split()) + add - remove)
```

Also note:
```python
>>> ''.split()
[]
>>> ''.split(' ')
['']
```
so to split on whitespace, *not* providing a separator can be better than providing one.

**Comment by Gorash** [comment]
After discussion, only add and remove, no eval (avoid python code in xml) ;)

File: openerp/addons/base/ir/ir_ui_view.py (hunk 1 above)
Comment (Gorash) [inline openerp/addons/base/ir/ir_ui_view.py]: Yep ^^
I did not know the trick to split. Thanks

**Comment by wtaferner** [comment]
@Gorash 
I was about to ask you about the eval, but I was sure you will reconsider or implement a safe eval :-)

**Comment by wtaferner** [comment]
Another question, especially about the attribute ```attrs``` - do you see any chance to help inheriting the dict b/c rewriting the whole dict is kind of annoying...this PR is more about css classes I guess, but anyway...

**Comment by Gorash** [comment]
I have not finished thinking (t-attf-class, the data-selector for snippet...)

**Comment by wtaferner** [comment]
Hehe, good, keep on thinking as there is potential inside...mostly I was blocked by attrs until now, but the others are important too, but I was accepting as it offers a lot of freedom that I need to rewrite a lot...maybe others will join us as inheritance needs to rock ;-)

**Comment by antonylesuisse** [comment]
@odony eval or not / saas-6 or master ? The current use cas is class="" attribute for themes.

**Comment by antonylesuisse** [comment]
Merged in https://github.com/odoo/odoo/commit/f962b0f62831614137a9e1e509803bfa39be1110

**Pull request closed without merging**